docs/spec/lightclient/verification.md

```diff
@@ -844,9 +915,130 @@ If on the blockchain the validator set of the block at height
 
 
 
+## Supporting the IBC Relayer
 
+We might need to download and verify headers whose heights are smaller
+than lightStore.LatestVerified.Height. To formalize
+the solution, we start with the following methods for the LightStore
 
 
+```go
+func (ls LightStore) LatestPrevious(height Height) (LightBlock, bool) 
+```
+- Expected postcondition
+  - returns a light block *lb* that satisfies:
+      - *lb* is in lightStore
+      - *lb* is verified and not expired
+	  - *lb.Header.Height < height*
+	  - for all *b* in lightStore s.t. *b* is verified and not expired it
+        holds *lb.Header.Height >= b.Header.Height*
+   - or false in the second argument if
+    the LightStore does not contain such an *lb*.
+ 
+
+```go
+func (ls LightStore) MinVerified() (LightBlock, bool) 
+```
+- Expected postcondition
+  - returns a light block *lb* that satisfies:
+      - *lb* is in lightStore
+      - *lb* is verified
+	  - *lb.Header.Height* is minimal in the lightStore
+	  - **TODO:** according to this it might be expired (outside the
+        trusting period). I believe this is safe. Are there reasons we
+        should not do that?
+   - or false in the second argument if
+    the LightStore does not contain such an *lb*.
+ 
+If a height that is smaller than the smallest height in the lightstore
+is required, we check the hashes backwards. This is done with the
+following function:
+ 
+#### **[LCV-FUNC-BACKWARDS.1]**:
+```go
+func Backwards(primary PeerID, lightStore LightStore,
+                    targetHeight Height) (LightStore, Result) {
+  
+  lb,res = lightStore.MinVerified()
+  if res = false {
+    return (lightStore, ResultFailure)
+  }
+
+  latest := lb.Header
+  for i := lb.Header.height - 1; i >= targetHeight; i-- {
+    // here we download height-by-height. We might first download all
+	// headers down to targetHeight and then check them.
+    current := FetchLightBlock(primary,i)
+    if (hash(current) != latest.Header.LastBlockId) {
+      return (lightStore, ResultFailure)
+    }
+	else {
+	  lightStore.Update(current, StateVerified)
```

Comment:
This is a new type? I guess the idea is to mark headers obtained by backward verification differently? Can they be used as a basis for forward verification?How do we reason about trusted period of those headers?

Reply:
It might make sense to introduce a new type here. Even if we do not distinguish now, at a later points we might find it useful.  
I guess we should discuss whether they should be treated in a special way. My current understanding is that if we go backwards from a trusted light block, and all hashes match, the trust extends to these light blocks as well. But perhaps I miss something.